From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Sat, 21 May 2022 10:20:10 +0100
Subject: MAXPATHLENectomy

Work around use of MAXPATHLEN via approved mechanisms, as not all
platforms have MAXPATHLEN (e.g., Windows and Hurd), and its use is
deprecated.
---
 src/core/chuck.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/core/chuck.cpp b/src/core/chuck.cpp
index 99c01e5..10d2d53 100644
--- a/src/core/chuck.cpp
+++ b/src/core/chuck.cpp
@@ -58,7 +58,6 @@
 #include <sys/param.h>   // added 1.3.0.0
 #else
 #include <direct.h>      // added 1.3.0.0
-#define MAXPATHLEN (255) // addec 1.3.0.0
 #endif // #ifndef __PLATFORM_WIN32__
 
 
@@ -503,11 +502,11 @@ bool ChucK::initCompiler()
     }
 
     std::string cwd;
-    char cstr_cwd[MAXPATHLEN];
+    char *cstr_cwd;
     
     // figure out current working directory (added 1.3.0.0)
     // is this needed for current path to work correctly?!
-    if( getcwd(cstr_cwd, MAXPATHLEN) == NULL )
+    if( (cstr_cwd = getcwd(NULL, 0)) == NULL )
     {
         // uh...
         EM_log( CK_LOG_SEVERE, "error: unable to determine current working directory!" );
@@ -516,6 +515,7 @@ bool ChucK::initCompiler()
     {
         // make c++ string
         cwd = std::string(cstr_cwd);
+        free(cstr_cwd);
         // add trailing "/"
         cwd += '/';
         // deferring this step until later, and only for Windows
